### PR TITLE
Create an equivalent API spec for API-endpoints.md file

### PR DESCRIPTION
**Pull request by lehphyro**
Add an OpenAPI file for `API-endpoints.md`, shall we delete the markdown file as well? It's possible to show an equivalent overview using IntelliJ.

### CODE REVIEW

File: questions.md [hunk 1]
```diff
@@ -36,7 +36,8 @@
 2. How would you have implemented the `getNextTierPromotionTime(Customer)` method?
 3. What do you think about the other parts of the class? How would you refactor?
     
-# Questions about 'API-endpoints.md'
+# Questions about 'API-endpoints.md' (use your IDE's support for folding code to present all endpoints in a single page)
```
Comment (jjjasper) [inline questions.md:39]: For the comment about using your IDE's support for folding code to make sense, this should be changed to 

> Questions about the `openapi.yaml`

File: questions.md [hunk 2]
```diff
@@ -36,7 +36,8 @@
...
+1. The business analyst has decided to help the team by supplying an API spec, what do you think about its design?
```
Comment (jjjasper) [inline questions.md:40]: There are multiple lines of questioning possible.

**Review by jjjasper — COMMENTED** [review]
Re: Should we delete the `endpoints.md` file?
I really like the way the blatant ugliness of the proposed API is pushed in your face by the minimalistic, just bullets, presentation of the endpoints. It also fits the story of the "not so technical BA" who is "trying to help reduce the discussions during refinement".

File: questions.md (hunk 1 above)
Comment (lehphyro) [reply questions.md:39]: Done.

File: questions.md (hunk 2 above)
Comment (lehphyro) [reply questions.md:40]: Yeah but we show this file to candidates during interviews, so we need to list them out.

**Comment by lehphyro** [comment]
I'm going to be off for two weeks, so please feel free to make any changes and merge.

**Review by jjjasper — APPROVED** [review]

**Review by ugurkaya88 — APPROVED** [review]

File: questions.md [hunk 3]
```diff
@@ -36,7 +36,8 @@
 2. How would you have implemented the `getNextTierPromotionTime(Customer)` method?
 3. What do you think about the other parts of the class? How would you refactor?
     
-# Questions about 'API-endpoints.md'
+# Questions about 'openapi.yaml' (use your IDE's support for folding code to present all endpoints in a single page)
+1. The business analyst has decided to help the team by supplying an API spec, what do you think about its design?
```
[the remark this answers is not in the document]
Comment (ugurkaya88) [reply questions.md:40]: That's different, one is more abstract / a requirement, one is more like an implementation / the solution itself; isn't it? :)

**Pull request merged**